### PR TITLE
chore(tests): Upgrades Jasmine and Karma

### PR DESCRIPTION
**Pull request by mrclay**
Also updates karma & jasmine (2.0 eases async testing considerably).

(Needed for testing #8350)

### CODE REVIEW

**Comment by mrclay** [comment]
cc @ewinslow

**Comment by ewinslow** [comment]
Failures?

**Comment by ewinslow** [comment]
@mrclay `sh: 1: karma: not found` got time to debug this? Would love to get this in.

**Comment by mrclay** [comment]
Really not sure where to start here

**Comment by ewinslow** [comment]
Maybe "npm test" would work if "karma" were a local path like
"./node_modules/bin/karma"?

File: package.json [hunk 1]
```diff
@@ -13,7 +12,7 @@
   },
   "homepage": "https://github.com/Elgg/Elgg",
   "scripts": {
-    "test": "karma start js/tests/karma.conf.js --single-run"
+    "test": "node_modules/karma/bin/karma start js/tests/karma.conf.js --single-run"
```
Comment (ewinslow) [inline package.json]: Glad this worked. Kind of annoying, but I guess `npm test` abstracts away the boilerplate anyways so whatevs!

**Pull request merged**